app/controllers/admin/facilities_controller.rb

```diff
@@ -108,46 +107,11 @@ def parse_file
     @facilities = Facility.parse_facilities(@file_contents)
   end
 
-  def validate_at_least_one_facility
-    @errors << "Uploaded file doesn't contain any valid facilities" if @facilities.blank?
-  end
-
-  def validate_duplicate_rows
-    facilities_slice = @facilities.map { |facility|
-      facility.slice(:organization_name, :facility_group_name, :name) }
-    @errors << 'Uploaded file has duplicate facilities' if
-        facilities_slice.count != facilities_slice.uniq.count
-  end
-
-  def validate_facilities
-    @row_errors = []
-    row_num = 2
-    @facilities.each do |facility|
-      import_facility = Facility.new(facility)
-      @row_errors << [row_num, import_facility.errors.full_messages.to_sentence] if
-          import_facility.invalid?
-      row_num += 1
-    end
-    if @row_errors.present?
-      group_row_errors.each { |error| @errors << error }
-    end
+  def validate_facility_rows
+    @errors = Admin::FacilityCSVValidator.validate(@facilities).errors
```

Comment:
Curious, do we reject the entire CSV if any errors are found?

Reply:
Yes and we show the user something like:

![image](https://user-images.githubusercontent.com/16774200/82535071-f1432b00-9b63-11ea-9e16-2959adbb19a3.png)